admin/class-create-block-theme-admin.php

```diff
@@ -679,38 +696,50 @@ function create_admin_form_page() {
 								<br /><br />
 								<label>
-									<input value="clone" type="radio" name="theme[type]" class="regular-text code" onchange="document.getElementById('new_theme_metadata_form').removeAttribute('hidden');"/>
+									<input value="clone" type="radio" name="theme[type]" class="regular-text code" onchange="toggleForm( 'new_theme_metadata_form', false );"/>
 									<?php _e('Clone ', 'create-block-theme'); echo wp_get_theme()->get('Name'); ?><br />
 									<?php _e('[Create a new theme cloning the activated theme. The resulting theme will have all of the assets of the activated theme as well as user changes.]', 'create-block-theme'); ?>
 								</label>
 								<br /><br />
 							<?php endif; ?>
 							<label>
-								<input value="save" type="radio" name="theme[type]" class="regular-text code" onchange="document.getElementById('new_theme_metadata_form').setAttribute('hidden', null);" />
+								<input value="save" type="radio" name="theme[type]" class="regular-text code" onchange="toggleForm( 'new_theme_metadata_form', true );toggleForm( 'new_variation_metadata_form', true );" />
 								<?php _e('Overwrite ', 'create-block-theme'); echo wp_get_theme()->get('Name'); ?><br />
 								<?php _e('[Save USER changes as THEME changes and delete the USER changes.  Your changes will be saved in the theme on the folder.]', 'create-block-theme'); ?>
 							</label>
 							<br /><br />
 							<label>
-								<input value="blank" type="radio" name="theme[type]" class="regular-text code" onchange="document.getElementById('new_theme_metadata_form').removeAttribute('hidden', null);" />
+								<input value="blank" type="radio" name="theme[type]" class="regular-text code" onchange="toggleForm( 'new_theme_metadata_form', false );" />
 								<?php _e('Create blank theme ', 'create-block-theme'); ?><br />
 								<?php _e('[Generates a boilerplate "empty" theme inside of this site\'s themes directory.]', 'create-block-theme'); ?>
 							</label>
 							<br /><br />
+							<label>
+								<input value="variation" type="radio" name="theme[type]" class="regular-text code" onchange="toggleForm( 'new_variation_metadata_form', false );" />
+								<?php _e('Create a style variation ', 'create-block-theme'); ?><br />
+								<?php printf( esc_html__('[Saves user changes as a style variation of %1$s.]', 'create-block-theme'),  esc_html( wp_get_theme()->get('Name') ) ); ?>
+							</label>
+							<br /><br />
 
 							<input type="submit" value="<?php _e('Create theme', 'create-block-theme'); ?>" class="button button-primary" />
 
 						</div>
 					</div>
 					<div id="col-right">
 						<div class="col-wrap">
+							<div hidden id="new_variation_metadata_form">
+								<label>
+									<?php _e('Variation Name (*):', 'create-block-theme'); ?><br />
+									<input placeholder="<?php _e('Variation Name', 'create-block-theme'); ?>" type="text" name="theme[variation]" class="large-text" />
```

Comment:
We could use HTML5 validation on the required fields like theme name and variation name, so the feedback is quicker for the user. Something like this:

```suggestion
									<input placeholder="<?php _e('Variation Name', 'create-block-theme'); ?>" type="text" name="theme[variation]" class="large-text" required />
```

We should still keep the server-side validation too, but with the above, it means the page doesn't need to reload. Although we'd need to remove the `required` attribute from the forms that are hidden, so they're not evaluated if they're not being used... Maybe client-side validation could be handled in a separate PR (it could be added to the new `toggleForm` function).

Reply:
yeah we would need to toggle the attributes, that's why I didn't add it. Looks like a good enhancement for a followup PR